### ISSUE TITLE
Extended Bolus for Meal Entry

### ISSUE THREAD

**Issue by bmozisek** [timomer/HAPP#41]
I'm going to go ahead and put this request in.  There are certain meals that I know I will need an extended bolus to cover.  Would it be possible to be able to tell HAPP that we are doing an Extended bolus so that it can accurately keep the IOB during this time period.

**Comment by timomer** [comment]
yep, sounds like a good idea. No eta but ill start to have a think of the logic and how this will be managed

**Comment by gempickfordwaugh** [comment]
Both extended option and multiwave option would be fabulous 😀

**Comment by timomer** [comment]
sure ;)

Hang in there, as I will need to talk to the OpenAPS team to see how the system will take such boluses or how HAPP will need to format them.

I'm guessing HAPP will need to break them down to multiple micro boluses will be the most simple process.

**Comment by cbruyndoncx** [comment]
i always use dual bolus too, this feature is pretty high on my list too

**Comment by timomer** [comment]
Hey all, can I confirm that a dual bolus example would be like this...

2u now, 1u in 30mins

in other words, two boluses where one is timed to happen at a later time

**Comment by cbruyndoncx** [comment]
a dual bolus is basically part right now and the rest spread out over a time period.

So a 50%/50% split of 5units over 2.5 hours =
Bolus 2.5Units
Square 2.5Units over 2.5 hours, or 1U/hr or 0.5U/30mins, is actually delivered in even smaller amounts so it's 0.5/6 every 5 mins. I think 5 mins is a typical value, though would need to doublecheck to be certain.

**Comment by timomer** [comment]
great thanks, "dual bolus" and "Extended Bolus" I'm assuming are the same thing

The second Extended part of the Bolus would be split into 5min micro boluses and passed to OpenAPS, I want to run that past the OpenAPS team first. I am having a think about how to capture this in the UI and how to allow a stopped Extended Bolus

**Comment by cbruyndoncx** [comment]
Related but not 100% the same.
Square bolus is an bolus extended over a longer period.
Dual bolus, is sometimes also framed as multi-wave, but you only take one action. Part of the bolus is immediate (like a regular normal bolus) and part is extended over a period.
Whenever you dual bolus, you could do the same in 2 actions with a normal bolus and a square straight after.

**Comment by timomer** [comment]
ok, im going to read up and be sure this is the same behavior for all pumps, ill list some examples here for feedback

**Comment by kews12** [comment]
Any progress on this? or suggestion how to have it accounted for? I just started using HAPP along with xdrip & slow carb absorption is my major issue (which I discovered thanks to your apps!)

**Comment by timomer** [comment]
Hey, sorry for the delayed reply.

I have added this as a task for the HAPP+, I still need to get basic functions up and running before looking at adv events like this. Summer is turning out to be super busy so ill be honest this will take longer than I like to complete. Bust super motivated to get on with stuff, I need to start booking out one day a week for this work 

Very pleased to hear the apps have helped!!

**Comment by jroth-haj** [comment]
Hi Tim, 
I'm also voting for this. Here is my summary: 

dual bolus = multi wave bolus
Example: 
x IE, 60% now/ 40% in minimal steps for yh

extended bolus = square bolus
Example:
x IE in minimal steps for the next y hours

**Comment by timomer** [comment]
Noted :)

Due to personal commitments I am on hold right now with any work on the redesign with HAPP+, but plan to pick things up again come winter

**Comment by CaroGo** [comment]
This should be one of the most important improvements  to this application.... I am looking forward to the next step.... And I am full of respect for the work that is done on this (and all the other) issues! Thank you so much! It really improves the lives of many people!

**Comment by timomer** [comment]
@CaroGo thanks, don't worry, this will be picked up again :)
